### ISSUE TITLE
Non-deterministic behavior when Run on Different GPU in different Machines

### ISSUE THREAD

**Issue by hontimzam** [tensorflow/tensorflow#55695]
Hello everyone, I would like to ask a determinism Issue with **Tensorflow version 1.15**

I tried to run some ML-model without tf.data.Dataset. (As i know there is some random in tf.data.Dataset) I also tried to add:
os.environ['TF_CUDNN_DETERMINISTIC'] = '1'
random.seed(1)
np.random.seed(1)
tf.set_random_seed(1)

The results is that I can reproduce the results with 100% similarity in the **SAME** machine. However, I cannot reproduce the results with 100% similarity in other machine.
Due to some issue in my old GPU, I cannot upgrade the **tensorflow** version, so we keep the version of 1.15
In our case, we try to run ML-model (Machine Learning - with Convolutional Neural Networks model) in two different Machines, one with the **GPU**:
**GeForce RTX 2070
NVIDIA-SMI 440.33.01    
Driver Version: 440.33.01    
CUDA Version: 10.2**

Another machine with the **GPU**:
**GeForce RTX 2080
NVIDIA-SMI 460.91.03    
Driver Version: 460.91.03    
CUDA Version: 11.2**  

The ML-results shows differently with these two machine. I hear about the difference of hard-ware version/**CUDA** version might cause the random results. May I ask that can anyone fix these issue? OR if the issue really cannot fix, which one has a bigger impact, difference of **GPU** hardware or **CUDA** version?

In other words, if the **GPUs** are the same, but **CUDAs** version are not the same, is it better than same **CUDA** with different **GPU**?

Thank you very much.

**Comment by tilakrayal** [comment]
Hello @hontimzam, 
We see that you are using tf version 1.5.0, 1.x is not actively supported, I request you,please update to latest v2.x and let us know if you are facing same issue.
Also please refer this [doc link](https://github.com/tensorflow/docs/blob/master/site/en/install/source.md#gpu) for the compatible CUDA and cudNN configurations for v1.5.0.Thanks!

**Comment by hontimzam** [comment]
Hello, 

> Hello @hontimzam, We see that you are using tf version 1.5.0, 1.x is not actively supported, I request you,please update to latest v2.x and let us know if you are facing same issue. Also please refer this [doc link](https://github.com/tensorflow/docs/blob/master/site/en/install/source.md#gpu) for the compatible CUDA and cudNN configurations for v1.5.0.Thanks!

**Comment by hontimzam** [comment]
Hello, I am sorry for the wrong click "closed" issue. For the tf version. I am sorry for the wrong information. we are using Tension flow version of **1.15.x** . In this case, do we able to solve the non-deterministic behavior on different GPU? We do not upgrade from 1.15.x to 1.2.x because of some compatible issue.

**Comment by tilakrayal** [comment]
Hello @hontimzam,
As suggested,1.x is not actively supported, I request you,please update to latest v2.x and let us know if you are facing same issue.
Also please refer this [doc link](https://github.com/tensorflow/docs/blob/master/site/en/install/source.md#gpu) for the compatible CUDA and cudNN configurations for v1.15.0.Thanks!
